app/api/db/models/lookup.py

```diff
@@ -0,0 +1,186 @@
+#
+# Classes for handling database lookup tables.
+#
+
+from typing import Any, Dict
+
+import api.logging
+
+logger = api.logging.get_logger(__name__)
+
+
+class LookupTable:
```

Comment:
This is 100% verbatim from the Mass approach. I tried to think of another approach to creating a static list of values, but kept coming back around to this approach. Open to suggestions if there's another common pattern. I do like that the static values live in the DB which makes direct SQL queries a bit more straightforward from past experience.

Reply:
what are the benefits of having a lookup tables rather than an enum type in SQL?

Reply:
I was not aware of the enum type in SQL (why didn't we use that in Mass?) - just sorta assumed it didn't exist and never checked. Let me see how that works.

Reply:
Looks like you can alter enum values in postgres as well: https://blog.yo1.dog/updating-enum-values-in-postgresql-the-safe-and-easy-way/ but not sure if it's worth the hassle.

@lorenyu  Any thoughts on how we should progress here? That not recognizing an alteration problem would need some solution to make this feel like a good approach. Should we stick with the lookup table approach, or dive further into enums (might be out of scope for the time I have to get this done)?

Reply:
I did find someone who figured out auto-migrating for enums: https://github.com/dw/alembic-autogenerate-enums/blob/master/alembic_autogenerate_enums.py

Reply:
I'm going to push forward as-is, but I did put together a quick branch which shows enums with the above library. The down migration is misleading however as it does nothing.
https://github.com/navapbc/template-application-flask/blob/chouinar/enum-example/app/api/db/models/user_models.py

Creates the following migration: https://github.com/navapbc/template-application-flask/blob/chouinar/enum-example/app/api/db/migrations/versions/2022_09_21_10_44_26_f9e1862f9c52_create_user_tableagain.py and with the adjusted migration script, and an adjustment to the enum produces: https://github.com/navapbc/template-application-flask/blob/chouinar/enum-example/app/api/db/migrations/versions/2022_09_21_13_21_15_389e7f257581_add_another_enum_val.py

Reply:
@chouinar sounds good to proceed as-is for now.

re: the other branch, do we still need "sync_lookup_tables" with enums?

Reply:
No, wouldn't need that, I just switched to the bare minimum to get it working, but all the lookup table specific logic isn't necessary.

Reply:
Bringing in the convo from Slack, I think we should keep it simple and get rid of all this boilerplate code and just have Role be a TEXT column and in the application code it could be a class perhaps an [Enum](https://docs.python.org/3/library/enum.html) type in Python (looks like pydantic also just leverages the existing python enum as well)

Reply:
If you want to do that in a separate PR lemme know I could approve this one for now

Reply:
I went ahead and switched to having the validation purely in the API/Pydantic layer. The DB just has it as a text column.

Note that I did also look into the check constraint for validating the text field, but alembic also can't auto-detect changes to it, so it ends up being effectively the same a the enum field.